### PR TITLE
Unify the `LightningOptimizer` and `LiteOptimzier`

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

After #11444, both implementations can be merged.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @justusschock @awaelchli @rohitgr7 @carmocca @akihironitta

### CODE REVIEW

File: tests/lite/test_wrappers.py [hunk 1]
```diff
@@ -133,20 +133,11 @@ def test_lite_dataloader_device_placement(src_device, dest_device):
     assert torch.equal(batch1["data"], torch.tensor([2, 3], device=dest_device))
 
 
-def test_lite_optimizer_wraps():
```
Comment (carmocca) [inline tests/lite/test_wrappers.py:136]: This test is not specific to Lite so I moved it

File: pytorch_lightning/core/optimizer.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def optimizer(self) -> Optimizer:
         return self._optimizer
 
     @classmethod
-    def _to_lightning_optimizer(
```
Comment (carmocca) [inline pytorch_lightning/core/optimizer.py:61]: Renamed since it's also used in Lite now

**Comment by carmocca** [comment]
Re-marked as draft since I need to split out the profiler changes first

**Review by tchaton — APPROVED** [review]
LGTM !

File: pytorch_lightning/core/optimizer.py (hunk 2 above)
Comment (kaushikb11) [reply pytorch_lightning/core/optimizer.py:65]: But the variable names inside the method mentions lightning optimizer. Maybe need to update them too?

Comment (carmocca) [reply pytorch_lightning/core/optimizer.py:65]: What do you mean? The output is still a LightningOptimizer.

I just renamed it because it was weird to call

`_LiteOptimizer._to_lightning_optimizer`

**Review by kaushikb11 — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/lite/wrappers.py [hunk 3]
```diff
@@ -11,58 +11,24 @@
...
-    def __init__(self, optimizer: Optimizer, strategy: Strategy) -> None:
-        """LiteOptimizer is a thin wrapper around the :class:`~torch.optim.Optimizer` that delegates the optimizer
-        step calls to the strategy plugin.
-
-        The underlying wrapped optimizer object can be accessed via the property :attr:`optimizer`.
-
-        Args:
-            optimizer: The optimizer to wrap
-            strategy: Reference to the strategy for handling the optimizer step
-        """
```
Comment (awaelchli) [inline pytorch_lightning/lite/wrappers.py:42]: can we keep the docstring please? I really think the one from LightningOptimizer sucks.

File: pytorch_lightning/lite/wrappers.py [hunk 4]
```diff
@@ -11,58 +11,24 @@
...
-        # `__del__` is skipped in case the optimizer has implemented custom destructor logic which we would
-        # not want to call on destruction of the `_LiteOptimizer
-        self.__dict__ = {k: v for k, v in optimizer.__dict__.items() if k not in ("state_dict", "step", "__del__")}
```
Comment (awaelchli) [inline pytorch_lightning/lite/wrappers.py:45]: what about state_dict?
